### ISSUE TITLE
The top bar More Menu should be a unique ARIA menu

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#12505]
The More Menu in the top toolbar is made of a collection of ARIA menus, each one with its own menu items. Plugins can add their own menu items in the "Plugins" sub-menu.

However, this menu (screenshot on the left) should work in the same way as the Block "More options" menu (screenshot on the right):

<img width="591" alt="screenshot 2018-12-01 at 15 51 38" src="https://user-images.githubusercontent.com/1682452/49330240-26cd1080-f58c-11e8-83ef-0202b8b2d13f.png">

I'm not referring to the visuals: it's about semantics, accessibility, and keyboard interaction:
- the whole menu should be a single ARIA menu 
- groups can be made using an ARIA `role="group"`, see the W3C example from the ARIA authoring practices: https://www.w3.org/TR/wai-aria-practices-1.1/examples/menubar/menubar-2/menubar-2.html where "Style/Color" and "Size" have groups
- each group can use a separator and should be labelled with `aria-labelledby` referencing the visible title
- most importantly: this way, arrows navigation will work across the whole menu, while currently it only works within each single group

Basically, the `MenuGroup` component should be just a `group` of menu items and not render a `NavigableMenu`. Then, the groups should be wrapped by an ARIA menu, i.e. a single `NavigableMenu`.

Some care should be used for the visible "titles", as the items container in an ARIA menu can only be menuitems or separators child elements of the containing menu so the title should be removed from the accessibility tree.

This will probably require some small refactoring and evaluating how the changes impact the current usage of these components.

Reference:
https://www.w3.org/TR/wai-aria-practices-1.1/#menu

**Comment by Firestorm980** [comment]
@afercia and @tofumatt I'm taking a stab at this one.

**Comment by Firestorm980** [comment]
New PR for this one: https://github.com/WordPress/gutenberg/pull/13059

**Comment by gziolo** [comment]
Work is going to continue in #14843.
